File: src/main/java/com/pebble/baseAuth/persistence/UserJpaRepository.java
package com.pebble.baseAuth.persistence;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserJpaRepository extends JpaRepository<UserEntity, Long> {

    boolean existsByUsernameAndDeletedAtIsNull(String username);

    Optional<UserEntity> findByUsernameAndDeletedAtIsNull(String username);

    Optional<UserEntity> findByProviderAndProviderIdAndDeletedAtIsNull(String provider, String providerId);

}

File: auth-module/src/main/kotlin/com/pebble/basicAuth/persistence/UserJpaRepository.kt
package com.pebble.basicAuth.persistence

import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface UserJpaRepository : JpaRepository<UserEntity, Long> {

    fun existsByUsernameAndDeletedAtIsNull(username: String): Boolean

    fun findByUsernameAndDeletedAtIsNull(username: String): Optional<UserEntity>

    fun findByEmailAndDeletedAtIsNull(email: String): Optional<UserEntity>

    fun existsByEmailAndDeletedAtIsNull(email: String): Boolean

    fun findByProviderAndProviderIdAndDeletedAtIsNull(provider: String, providerId: String): Optional<UserEntity>

}
